example/notify_inval_entry.c

```diff
@@ -253,14 +257,33 @@ static void* update_fs_loop(void *data) {
     struct fuse_session *se = (struct fuse_session*) data;
     char *old_name;
 
-    while(1) {
+
+    while(!fuse_session_exited(se)) {
         old_name = strdup(file_name);
         update_fs();
+
         if (!options.no_notify && lookup_cnt) {
-            if(options.only_expire) {
-                assert(fuse_lowlevel_notify_expire_entry
-                   (se, FUSE_ROOT_ID, old_name, strlen(old_name), FUSE_LL_EXPIRE_ONLY) == 0);
-            } else {
+            if(options.only_expire) { // expire entry
+                int ret = fuse_lowlevel_notify_expire_entry
+                   (se, FUSE_ROOT_ID, old_name, strlen(old_name));
+
+                // no kernel support
+                if (ret == -ENOSYS) {
+                    printf("fuse_lowlevel_notify_expire_entry not supported by kernel\n");
+                    printf("Exiting...\n");
+                    se->error = ret;
+
+                    fuse_session_exit(se);
+                    // needed to correctly exit fuse_session_loop()
```

Comment:
Suggest to phrase this as `// Make sure to exit now, rather than on next request from userspace`. There is nothing incorrect about not sending SIGPIPE.